aries-site/src/examples/templates/dashboards/content/chartCards/CostByService.js

```diff
@@ -0,0 +1,169 @@
+import { useContext, useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Box, Grid, Meter, ResponsiveContext, ThemeContext } from 'grommet';
+import { parseMetricToNum } from 'grommet/utils';
+import { ChartCard, Legend, Measure } from '../../components';
+
+const MOCK_DATA = require('../../../../../data/mockData/consumption.json');
+
+const current = new Date();
+const endDate = current;
+const beginDate = new Date(new Date().setDate(current.getDate() - 365));
+
+const defaultWindow = { begin: beginDate, end: endDate };
+
+const REPORT_WINDOW_MAP = {
+  'Last 30 Days': 30,
+  'Last Year': 365,
+};
+
+export const CostByService = ({ period }) => {
+  const [values, setValues] = useState(null);
+  const [totalCost, setTotalCost] = useState(null);
+  const [reportWindow, setReportWindow] = useState(defaultWindow);
+  const consumptionData = MOCK_DATA.consumption;
+  const size = useContext(ResponsiveContext);
+  const theme = useContext(ThemeContext);
+  const chartColors = useMemo(
+    () =>
+      Object.entries(theme.global.colors)
+        .filter(([key]) => key.includes('graph'))
+        .map(([, value]) => value),
+    [theme.global.colors],
+  );
+
+  // Set reporting window
+  useEffect(() => {
+    setReportWindow(prevWindow => ({
+      ...prevWindow,
+      begin: new Date(
+        new Date().setDate(
+          prevWindow.end.getDate() - REPORT_WINDOW_MAP[period],
```

Comment:
Just making sure I'm following the logic. So by default we will show previous year(365 days) but in the cost by service chart we are passing in a period of 30 days and so this effect will be called to affect the starting date?

Reply:
Correct. You got it.